openedx/core/djangoapps/content/learning_sequences/api/outlines.py

```diff
@@ -122,9 +123,11 @@ def get_course_outline(course_key: CourseKey) -> CourseOutlineData:
     # represented (so query CourseSection explicitly instead of relying only on
     # select_related from CourseSectionSequence).
     section_models = CourseSection.objects \
+        .prefetch_related('user_partition_groups') \
```

Comment:
Are there any memory concerns with this pre-fetching? Ah - nm - just checked COURSE_GROUPS_COURSEUSERGROUPPARTITIONGROUP and it has only ~1,650 rows.

Reply:
Yeah, the number of partition groups actually associated with any particular course should be very low, especially since almost none are associated at the Section or Sequence layer.

Reply:
👍🏼